addons/pos_restaurant/models/pos_config.py

```diff
@@ -39,6 +39,13 @@ def get_tables_order_count(self):
         """         """
         self.ensure_one()
```

Comment:
suggestion for an alternative with less total queries and an intermediary map, and no manual filtering of inactive records:

```python
    def get_tables_order_count(self):
        self.ensure_one()
        tables = self.env['restaurant.table'].search([('floor_id.pos_config_id', 'in', self.ids)])
        domain = [('state', '=', 'draft'), ('table_id', 'in', tables.ids)]

        order_stats = self.env['pos.order'].read_group(domain, ['table_id'], 'table_id')
        orders_map = dict((s['table_id'][0], s['table_id_count']) for s in order_stats)

        result = []
        for table in tables:
            result.append({'id': table.id, 'orders': orders_map.get(table.id, 0)})
        return result
```

Reply:
Thanks